src/jit/SimdRiscvInl.h

```diff
@@ -0,0 +1,1251 @@
+    vminu_vv = InstructionType::opivv | OPCODE(0x4),
+    vmseq_vv = InstructionType::opivv | OPCODE(0x18),
+    vmsle_vv = InstructionType::opivv | OPCODE(0x1D),
+    vmsleu_vv = InstructionType::opivv | OPCODE(0x1C),
+    vmslt_vv = InstructionType::opivv | OPCODE(0x1B),
+    vmslt_vx = InstructionType::opivx | OPCODE(0x1B),
+    vmsltu_vv = InstructionType::opivv | OPCODE(0x1A),
+    vmsne_vi = InstructionType::opivi | OPCODE(0x19),
+    vmsne_vv = InstructionType::opivv | OPCODE(0x19),
+    vmul_vv = InstructionType::opmvv | OPCODE(0x25),
+    vmv_sx = InstructionType::opmvx | OPCODE(0x10),
+    vmv_vi = InstructionType::opivi | OPCODE(0x17),
+    vmv_vv = InstructionType::opivv | OPCODE(0x17),
+    vmv_vx = InstructionType::opivx | OPCODE(0x17),
+    vmv_xs = InstructionType::opmvv | OPCODE(0x10),
+    vor_vv = InstructionType::opivv | OPCODE(0xA),
+    vredmaxu_vs = InstructionType::opmvv | OPCODE(0x6),
+    vredminu_vs = InstructionType::opmvv | OPCODE(0x4),
+    vredsum_vs = InstructionType::opmvv | OPCODE(0x0),
+    vrgather_vv = InstructionType::opivv | OPCODE(0xC),
+    vrsub_vi = InstructionType::opivi | OPCODE(0x3),
+    vsadd_vv = InstructionType::opivv | OPCODE(0x21),
+    vsaddu_vv = InstructionType::opivv | OPCODE(0x20),
+    vsext_vf2 = InstructionType::opmvv | OPCODE(0x12) | (0x7 << 15),
+    vslidedown_vi = InstructionType::opivi | OPCODE(0xF),
+    vsll_vi = InstructionType::opivi | OPCODE(0x25),
+    vsll_vx = InstructionType::opivx | OPCODE(0x25),
+    vsra_vi = InstructionType::opivi | OPCODE(0x29),
+    vsra_vx = InstructionType::opivx | OPCODE(0x29),
+    vsrl_vi = InstructionType::opivi | OPCODE(0x28),
+    vsrl_vx = InstructionType::opivx | OPCODE(0x28),
+    vssub_vv = InstructionType::opivv | OPCODE(0x23),
+    vssubu_vv = InstructionType::opivv | OPCODE(0x22),
+    vsub_vv = InstructionType::opivv | OPCODE(0x2),
+    vwmul_vv = InstructionType::opmvv | OPCODE(0x3B),
+    vxor_vi = InstructionType::opivi | OPCODE(0xB),
+    vxor_vv = InstructionType::opivv | OPCODE(0xB),
+    vzext_vf2 = InstructionType::opmvv | OPCODE(0x12) | (0x6 << 15),
+};
+
+enum OperandTypes : uint32_t {
+    rn_imm = 1 << 1,
+    rm_imm = 1 << 2,
+    rn_gpreg = 1 << 3,
+    rm_gpreg = 1 << 4,
+    rd_gpreg = 1 << 5
+};
+}; // namespace SimdOp
+
+static void simdEmitVsetivli(struct sljit_compiler* compiler, sljit_s32 type, sljit_ins vlmul)
+{
+    uint32_t elem_size = (uint32_t)(((type) >> 18) & 0x3f);
+    uint32_t avl = (uint32_t)1 << (4 - elem_size);
+
+    uint32_t opcode = VSETIVLI | (6 << 7) | (elem_size << 23) | (vlmul << 20) | (avl << 15);
+    sljit_emit_op_custom(compiler, &opcode, sizeof(uint32_t));
+}
+
+static void simdEmitOp(sljit_compiler* compiler, uint32_t opcode, sljit_s32 rd, sljit_s32 rn, sljit_s32 rm, uint32_t optype = 0)
+{
+    rd = sljit_get_register_index((optype & SimdOp::rd_gpreg) ? SLJIT_GP_REGISTER : SLJIT_SIMD_REG_128, rd);
+    if (!(optype & SimdOp::rn_imm) && !(optype & SimdOp::rn_gpreg) && rn >= SLJIT_VR0) {
```

Comment:
Can rn be zero in this case? If not, it should be an assert. Similar cases below.

Reply:
`(rn == SLJIT_VR0) >= SLJIT_VR0`, so it is still true without an extra >=